packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx

```diff
@@ -37,13 +37,13 @@ export function useDocumentVersions(props: DocumentPerspectiveProps): DocumentPe
   const {documentId} = props
 
   const {data: bundles} = useBundles()
-  const publishedId = getPublishedId(documentId, documentId.includes('.'))
+  const publishedId = getPublishedId(documentId)
 
   const documentPreviewStore = useDocumentPreviewStore()
 
   const observable = useMemo(() => {
     return documentPreviewStore
-      .unstable_observeDocumentIdSet(`sanity::versionOf("${publishedId}")`)
+      .unstable_observeDocumentIdSet(`_id in path("versions.**") && _id match '*.${publishedId}'`)
```

Comment:
Nice combination of `path` and `match` 🚀

Reply:
I can't take credit for this since I asked @bjoerge for help with it 🙏